tests/monotouch-test/Compression/ThoroughCompressionStreamTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+
+using Foundation;
+using Compression;
+using ObjCRuntime;
+
+using NUnit.Framework;
+
+
+namespace MonoTouchFixtures.Compression {
+	
+	// Special test case that tests the compression class against files that
+	// have been compressed by apple tools. This ensures that we can read
+	// the data from a file created by an outside tool.
+	// The tests have been moves outside the CompressionStreamTest because
+	// those tests are exactly the same ones as the DeflateStream ones while
+	// this are just for the CompressionStream implementation.
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class ThoroughCompressionStreamTest
+	{
+
+#if MONOMAC
+ 		string zipCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "Contents/Resources/compressed_zip");
+ 		string lzmaCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "Contents/Resources/compressed_lzma");
+ 		string lz4CompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "Contents/Resources/compressed_lz4");
+ 		string lzeCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "Contents/Resources/compressed_lze");
+ 		string lzeUncompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "Contents/Resources/uncompressed.txt");
+#else
+ 		string zipCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "compressed_zip");
+ 		string lzmaCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "compressed_lzma");
+ 		string lz4CompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "compressed_lz4");
+ 		string lzeCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "compressed_lze");
+ 		string lzeUncompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "uncompressed.txt");
```

Comment:
could be simplified with having the path logic (once) inside the test

Reply:
Yes, I had that in my todo, remember [WIP] I was simply doing ugly tests :)